pkg/cmd/pipeline/logs_test.go

```diff
@@ -648,264 +303,294 @@ func TestLogs_interactive_ask_all_last_run(t *testing.T) {
-	if err != nil {
-		t.Error("Expecting no error")
-	}
-}
-
-func TestLogs_have_one_get_one(t *testing.T) {
-	clock := clockwork.NewFakeClock()
-
-	cs, _ := test.SeedTestData(t, pipelinetest.Data{
-		Pipelines: []*v1alpha1.Pipeline{
-			tb.Pipeline(pipelineName, ns,
-				// created  15 minutes back
-				cb.PipelineCreationTimestamp(clock.Now().Add(-15*time.Minute)),
-			),
-		},
-		PipelineRuns: []*v1alpha1.PipelineRun{
-
-			tb.PipelineRun(prName, ns,
-				cb.PipelineRunCreationTimestamp(clock.Now().Add(-10*time.Minute)),
-				tb.PipelineRunLabel("tekton.dev/pipeline", pipelineName),
-				tb.PipelineRunSpec(pipelineName),
-				tb.PipelineRunStatus(
-					tb.PipelineRunStatusCondition(apis.Condition{
-						Status: corev1.ConditionTrue,
-						Reason: resources.ReasonSucceeded,
-					}),
-					// pipeline run started 5 minutes ago
-					tb.PipelineRunStartTime(clock.Now().Add(-5*time.Minute)),
-					// takes 10 minutes to complete
-					cb.PipelineRunCompletionTime(clock.Now().Add(10*time.Minute)),
-				),
-			),
+		{
+			name:      "basic interaction5",
+			limit:     5,
+			last:      true,
+			namespace: ns,
+			input:     cs,
+			prompt: htest.PromptTest{
+				CmdArgs: []string{},
+				Procedure: func(c *goexpect.Console) error {
+					if _, err := c.ExpectString("Select pipeline:"); err != nil {
+						return err
+					}
+
+					if _, err := c.SendLine(string(terminal.KeyArrowDown)); err != nil {
+						return err
+					}
+
+					if _, err := c.ExpectString("output-pipeline"); err != nil {
+						return err
+					}
+
+					if _, err := c.SendLine(string(terminal.KeyEnter)); err != nil {
+						return err
+					}
+
+					if _, err := c.ExpectString("Select pipelinerun:"); err == nil {
+						return errors.New("unexpected error")
+					}
+
+					if _, err := c.ExpectEOF(); err != nil {
+						return err
+					}
+
+					return nil
+				},
+			},
 		},
-		Namespaces: []*corev1.Namespace{
-			{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: ns,
+		{
+			name:      "basic interaction6",
+			limit:     5,
+			last:      true,
+			namespace: ns,
+			input:     cs,
+			prompt: htest.PromptTest{
+				CmdArgs: []string{pipelineName},
+				Procedure: func(c *goexpect.Console) error {
+					if _, err := c.ExpectString("output-pipeline"); err == nil {
+						return errors.New("unexpected error")
+					}
+
+					if _, err := c.ExpectEOF(); err != nil {
+						return err
+					}
+
+					return nil
 				},
 			},
 		},
-	})
-
-	tests := []promptTest{
 		{
-			name:    "basic interaction",
-			cmdArgs: []string{pipelineName},
-
-			procedure: func(c *goexpect.Console) error {
-				if _, err := c.ExpectString("output-pipeline"); err == nil {
-					return errors.New("unexpected error")
-				}
-
-				if _, err := c.ExpectEOF(); err != nil {
-					return err
-				}
-
-				return nil
+			name:      "basic interaction7",
+			limit:     5,
+			last:      false,
+			namespace: ns,
+			input:     cs2,
```

Comment:
`cs2` seems to be the exact same as `cs`. Is it needed?

Reply:
the difference between cs and cs2 is namespace.
(cs uses "ns", cs2 uses ns)

so cs2 is used for another namespace test case!
(I agree it cause a bit confuse :sweat_smile:)

Reply:
Yes, good catch! I should've caught this as I added this 😄